dbt/adapters/athena/session.py

```diff
@@ -11,3 +23,244 @@ def get_boto3_session(connection: Connection) -> boto3.session.Session:
         region_name=connection.credentials.region_name,
         profile_name=connection.credentials.aws_profile_name,
     )
+
+
+class AthenaSparkSessionManager:
+    """
+    A helper class to manage Athena Spark Sessions.
+    """
+
+    def __init__(self, credentials: Any, timeout: int, polling_interval: float, engine_config: Dict[str, int]) -> None:
+        """
+        Initialize the AthenaSparkSessionManager instance.
+
+        Args:
+            credentials (Any): The credentials to be used.
+            timeout (int): The timeout value in seconds.
+            polling_interval (float): The polling interval in seconds.
+            engine_config (Dict[str, int]): The engine configuration.
+
+        """
+        self.credentials = credentials
+        self.timeout = timeout
+        self.polling_interval = polling_interval
+        self.engine_config = engine_config
+        self.lock = threading.Lock()
+
+    @cached_property
+    def spark_threads(self) -> int:
+        """
+        Get the number of Spark threads.
+
+        Returns:
+            Any: The number of Spark threads. If not found in the profile, returns the default thread count.
+        """
+        if not self.credentials.spark_threads:
+            LOGGER.debug(
+                f"""Spark threads not found in profile. Got: {self.credentials.spark_threads}.
+                Using default count: {DEFAULT_THREAD_COUNT}"""
+            )
+            return DEFAULT_THREAD_COUNT
+        return int(self.credentials.spark_threads)
+
+    @cached_property
+    def spark_work_group(self) -> str:
+        """
+        Get the Spark work group.
+
+        Returns:
+            Any: The Spark work group. Raises an exception if not found in the profile.
+        """
+        if not self.credentials.spark_work_group:
+            raise DbtRuntimeError(f"Expected spark_work_group in profile. Got: {self.credentials.spark_work_group}")
+        return str(self.credentials.spark_work_group)
+
+    @cached_property
+    def athena_client(self) -> Any:
+        """
+        Get the AWS Athena client.
+
+        This function returns an AWS Athena client object that can be used to interact with the Athena service.
+        The client is created using the region name and profile name provided during object instantiation.
+
+        Returns:
+            Any: The Athena client object.
+
+        """
+        return boto3.session.Session(
```

Comment:
Can you adapt this client as we do in the main adapter code?
We miss are missing to pass the boto3 configs for example.
Also, we need to be sure that different types of authentication works, not everyone can use profiles for example.

Reply:
I tried using that but I unfortunately don't know how to pass that `connection` object. Ideally I want to reuse the `AthenaConnectionManager`  but I am not sure if I can do something like this 

```
conn_man = AthenaConnectionManager()
get_boto3_session(conn_man.open())
```

Can you recommend how I'm supposed to modify this to be compatible with that get_boto3_session?

Reply:
I don't think that works - as when working with spark we have another connection handler.

In the end what we need to pass to **get_boto3_session** are the credentials, we can consider to refactor **get_boto3_session** to accept a `crendetials` config, not a connection. 

Alternatively we can create a new get_session, if you do so be sure to pass credentials and pass also get_boto3_config - where config is a passed to the session that is created.

Reply:
I have added a function `get_boto3_session_from_credentials`. I didn't want to refactor the existing method since it was used in many places. 

I have also passed the get_boto3_config to the client being created